diff --git a/oak-core/src/main/java/org/apache/jackrabbit/oak/stats/Clock.java b/oak-core/src/main/java/org/apache/jackrabbit/oak/stats/Clock.java
index a0b6b7d..6d6b3e8 100644
--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/stats/Clock.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/stats/Clock.java
@@ -168,7 +168,8 @@
         private long ns = System.nanoTime();
         @Override
         public synchronized long getTime() {
-            long nsIncrease = Math.max(System.nanoTime() - ns, 0); // >= 0
+            long nowns = System.nanoTime();
+			long nsIncrease = Math.max(nowns - ns, 0); // >= 0
             long msIncrease = nsIncrease / NS_IN_MS;
             if (ACCURATE_CLOCK_GRANULARITY > 1) {
                 msIncrease -= msIncrease % ACCURATE_CLOCK_GRANULARITY;
@@ -177,7 +178,7 @@
             long now = ms + msIncrease;
             if (now > ms + SYNC_INTERVAL) {
                 ms = SIMPLE.getTime();
-                ns = System.nanoTime();
+                ns = nowns;
                 // Check whether the system time jumped ahead or back
                 // from what we'd expect based on the nanosecond interval.
                 // If the jump was small, it was probably caused by low
