diff --git a/src/main/java/org/apache/commons/compress/compressors/bzip2/BZip2CompressorInputStream.java b/src/main/java/org/apache/commons/compress/compressors/bzip2/BZip2CompressorInputStream.java
index 1de731d..b1d20d9 100644
--- a/src/main/java/org/apache/commons/compress/compressors/bzip2/BZip2CompressorInputStream.java
+++ b/src/main/java/org/apache/commons/compress/compressors/bzip2/BZip2CompressorInputStream.java
@@ -521,7 +521,8 @@
                 }
                 zvec = (zvec << 1) | zj;
             }
-            nextSym = perm[zt][zvec - base[zt][zn]];
+            int[] perm_zt = perm[zt];
+			nextSym = perm_zt[zvec - base[zt][zn]];
         }
 
         while (true) {
